sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/TokenRefreshOptions.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.communication.common;
+
+/**
+ * Options for refreshing CommunicationTokenCredential
+ */
+public class TokenRefreshOptions {
+    private final TokenRefresher tokenRefresher;
+    private final boolean refreshProactively;
+
+    /**
+     * Creates a TokenRefreshOptions object
+     * 
+     * @param tokenRefresher the token refresher to provide capacity to fetch fresh token
+     * @param refreshProactively when set to true, turn on proactive fetching to call
+     *                           tokenRefresher before token expiry by minutes set
+     *                           with setCallbackOffsetMinutes or default value of
+     *                           two minutes
+     */
+    public TokenRefreshOptions(TokenRefresher tokenRefresher, boolean refreshProactively) {
```

Comment:
As mentioned above, `token` can go here too. The comment can explain that `token` is an optional initial token.

Reply:
Done, thanks Dom